src/Mantasflowers.Services/Services/Order/OrderService.cs

```diff
@@ -52,5 +74,27 @@ public async Task<GetDetailedOrderResponse> GetDetailedOrderInfoAsync(Guid id)
 
             return detailedOrderResponse;
         }
+
+        public async Task<Domain.Entities.UserOrder> LinkUserToOrderAsync(Guid userId, Domain.Entities.Order order)
+        {
+            var userOrder = new Domain.Entities.UserOrder
+            {
+                UserId = userId,
+                Order = order
+            };
+
+            try
+            {
+                userOrder = await _unitOfWork.UserOrderRepository.CreateAsync(userOrder);
+                await _unitOfWork.SaveChangesAsync();
+            }
+            catch (Exception ex) when
+            (ex is DbUpdateException || ex is ArgumentNullException)
```

Comment:
What is `ArgumentNullException` for?

Reply:
Same as a previous comment. Have a look at: OrderRepository and UserOrderRepository.